### PR TITLE
[CWS] introduce a generic way to capture packets

### PR DESCRIPTION
**Pull request by safchain**
### What does this PR do?

This PR introduces a way to capture header of raw packets. Follow-ups PR will add the in-kernel filtering capability which will target reducing the performance impact.

### CODE REVIEW

File: pkg/security/secl/model/category.go [hunk 1]
```diff
@@ -40,11 +40,31 @@ func GetAllCategories() []EventCategory {
...
+	// Kernel
+	case
+		BPFEventType.String(),
+		SELinuxEventType.String(),
+		MMapEventType.String(),
+		MProtectEventType.String(),
+		PTraceEventType.String(),
+		UnloadModuleEventType.String(),
+		BindEventType.String(),
+		DNSEventType.String():
```
Comment (YoannGh) [inline pkg/security/secl/model/category.go:61]: This should be in the `NetworkCategory`

File: pkg/security/resolvers/tc/resolver.go [hunk 2]
```diff
@@ -25,25 +26,31 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/security/secl/model"
 )
 
-// NetDeviceKey is used to uniquely identify a network device
-type NetDeviceKey struct {
-	IfIndex          uint32
-	NetNS            uint32
+// ProgmanKey is used to uniquely identify a tc program
+type ProgmanKey struct {
```
Comment (YoannGh) [inline pkg/security/resolvers/tc/resolver.go:30]:
```suggestion
type ProgramKey struct {
```

File: pkg/security/ebpf/c/include/hooks/network/imds.h [hunk 3]
```diff
@@ -29,6 +29,8 @@ int classifier_imds_request(struct __sk_buff *skb) {
         send_event_with_size_ptr(skb, EVENT_IMDS, evt, offsetof(struct imds_event_t, body) + (pkt->payload_len & (IMDS_MAX_LENGTH - 1)));
     }
 
+    tail_call_to_classifier(skb, RAW_PACKET);
```
Comment (safchain) [inline pkg/security/ebpf/c/include/hooks/network/imds.h:32]: remove this tail call

File: pkg/security/resolvers/tc/resolver.go (hunk 2 above)
Comment (safchain) [reply pkg/security/resolvers/tc/resolver.go]: Done

File: pkg/security/secl/model/category.go (hunk 1 above)
Comment (safchain) [reply pkg/security/secl/model/category.go]: Done

**Review by hestonhoffman — APPROVED** [review]

File: pkg/security/secl/model/model.go [hunk 4]
```diff
@@ -587,3 +587,16 @@ func (dfh *FakeFieldHandlers) ResolveProcessCacheEntry(_ *Event) (*ProcessCacheE
...
+type TLSContext struct {
+	Version uint16 `field:"version"` // SECLDoc[version] Definition:`TLS version`
+}
+
+// RawPacketEvent represents a packet event
+type RawPacketEvent struct {
+	NetworkContext
+
+	TLSContext TLSContext `field:"tls"`     // SECLDoc[tls] Definition:`TLS context`
+	Payload    string     `field:"payload"` // SECLDoc[payload] Definition:`Payload of the l3/l4 packet`
```
Comment (spikat) [inline pkg/security/secl/model/model.go:601]: Not sure we should keep this one

File: pkg/security/serializers/serializers_base.go [hunk 5]
```diff
@@ -216,6 +216,20 @@ type BaseEventSerializer struct {
 	*CGroupContextSerializer    `json:"cgroup,omitempty"`
 }
 
+// TSLContextSerializer defines a tls context serializer
+// easyjson:json
+type TSLContextSerializer struct {
```
Comment (spikat) [inline pkg/security/serializers/serializers_base.go:221]: nit: TLS*

Comment (safchain) [reply pkg/security/serializers/serializers_base.go]: s/TSL/TLS/ done

**Review by spikat — APPROVED** [review]

**Review by YoannGh — APPROVED** [review]

**Pull request merged**